Source/WebInspectorUI/UserInterface/Views/ConsoleSourceMapMessageGroup.js

```diff
@@ -0,0 +1,124 @@
+        groupElement.group = this;
+        this.element = groupElement;
+
+        // TODO set consoleMessage to use WI.UIStringWI.UIString("Source Map loading errors")
+        let consoleMessage = new WI.ConsoleMessage(WI.mainTarget, WI.ConsoleMessage.MessageSource.Other, WI.ConsoleMessage.MessageLevel.Error, WI.UIString("Source Map loading errors"));
+
+        let consoleMessageView = new WI.ConsoleMessageView(consoleMessage);
+        consoleMessageView.render();
+
+        var titleElement = consoleMessageView.element;
+        this._titleElement = titleElement;
+        this._titleElement.classList.add(WI.LogContentView.ItemWrapperStyleClassName, "hidden");
+        this._titleElement._expandable = false;
+
+        this._boundClickHandler = this.toggle.bind(this);
+        this._titleElement.addEventListener("click", this._boundClickHandler);
+
+        groupElement.appendChild(titleElement);
+
+        var messagesElement = document.createElement("div");
+        this._messagesElement = messagesElement;
+        groupElement.appendChild(messagesElement);
+
+        return groupElement;
+    }
+
+    addMessageView(messageView)
+    {
+        var element = messageView.element;
+        element.classList.add(WI.LogContentView.ItemWrapperStyleClassName);
+
+        let messageBodyElement = element.getElementsByClassName("console-message-body")?.item(0);
+        messageBodyElement?.classList.remove("console-top-level-message");
+
+        this.append(element);
+        this._messageCount++;
+        this._renderMessageCount();
+        
+        if (this._messageCount > 1 && this._titleElement._expandable == false) {
+            this._titleElement._expandable = true;
+            this._titleElement.classList.add("expandable");
+            this._titleElement.classList.remove("hidden");
+            this._messagesElement.classList.add("console-sourcemap-failures-container");
+            this._titleElement.appendChild(this._messagesElement);
+        }
+    }
+
+    toggle()
+    {
+        if (this._titleElement.classList.contains("expanded"))
+            this._titleElement.classList.remove("expanded");
+        else
+            this._titleElement.classList.add("expanded");
+    }
+
+    // Private
+
+    _renderMessageCount()
+    {
+        let count = this._messageCount;
+
+        if (count <= 1) {
+            if (this._messageCountElement) {
+                this._messageCountElement.remove();
+                this._messageCountElement = null;
+            }
+            return;
+        }
+
+        if (!this._messageCountElement) {
+            this._messageCountElement = document.createElement("span");
+            this._messageCountElement.classList.add("errors-count");
+            this._titleElement.insertBefore(this._messageCountElement, this._titleElement.firstChild);
+        }
+
+        this._messageCountElement.textContent = Number.abbreviate(count);
```

Comment:
If opting to do the reparenting of the first console message into an always collapsible group, 
the `this._messageCountElement` can be created in the constructor (or in `render()`) and the count can be incremented in `addMessageView`, thus obviating the need for this entire `_renderMessageCount()` method.

Reply:
Addressed this in the new approach.